### ISSUE TITLE
Typos in HW 4

### ISSUE THREAD

**Issue by jmbejara** [jmbejara/comp-econ-sp18#39]
Here is a list of known typos and other improvements to make in HW 4. I haven't corrected these particular ones yet. If anyone notices any other typos, please let me know here.

- Histograph should be histogram
- Be clear about use the ASEC samples and not the regular monthly samples (from March)
- Be more clear about where we should be dropping missing values. Some people have had issues matching my numbers and the issue seems to be affected by when we choose to drop missing variables in our code (before or after dropping certain columns).

**Comment by jmbejara** [comment]
- Q10. I am using a strict inequality to compare real wages to quantiles. The problem description is not clear about this.

**Comment by jmbejara** [comment]
- In the HW, I have code in there that drops `UHRSWORKLY`, even though we are told to use it later. We should not drop `UHRSWORKLY`

**Comment by jmbejara** [comment]
- Q8. (Plot a histogram of the average weekly hours and the average annual hours worked. ) This should ask for a histogram of `UHRSWORKLY` and for a histogram of `annual_hours`

**Comment by jmbejara** [comment]
This is how I am subsetting based on GQ. I don't include GQ = 0 because the codebook says something about 0 being NIU. My comment in the HW should reflect this.

```python
# GQ = 0 for vacant units, 1 for Households, 2 for group quarters
df = df[df.GQ == 1]
```

**Comment by afgong** [comment]
In Q15, do you mean compute three correlations? So the correlation between `ave_wages` and `median_wages`, `ave_wages` and `employment`, and `median_wages` and `employment`?

**Comment by jmbejara** [comment]
There will be a matrix of correlations. The entries of the matrix will have the correlations between each combination of pairs. There is a single command for this.

**Comment by afgong** [comment]
In Q20, are we looking to space the bins like `[25, 30, 35, 40, 45, 50, 55]`? Also, is `educ_bins` supposed to correspond to the codebook values? So `educ_bins` should be a list of 5 elements?

**Comment by jmbejara** [comment]
- Age bins look right to me.
 - The numerical order of the education codes seems to match the ascending levels of education. For that reason, I use the binning function to group the education codes into my custom defined education groups. There are 5 education groups, so there should be 6 numbers in the list.

**Comment by afgong** [comment]
![screen shot 2018-05-02 at 12 27 27](https://user-images.githubusercontent.com/37759075/39539021-366c056c-4e04-11e8-94c0-e17bfb566ad2.png)

Is the graph from Q16 supposed to look something like this?

**Comment by afgong** [comment]
Also, for Q22, how do you remove the `average_wage` above the Bachelors_Degree, so that in the heatmap in the following question, it doesn't look like this: 

![screen shot 2018-05-02 at 13 09 07](https://user-images.githubusercontent.com/37759075/39541133-66656758-4e0a-11e8-8271-2dc33ac511df.png)

![screen shot 2018-05-02 at 13 11 37](https://user-images.githubusercontent.com/37759075/39541136-6900a086-4e0a-11e8-86a2-50d226183562.png)

**Comment by jmbejara** [comment]
This is what mine looks like:

![q16](https://user-images.githubusercontent.com/3770323/39540847-857bcbf6-4e09-11e8-9305-9ecbd4cb8c59.png)

This might help. Here I have run `df.describe()` at various points. 

At Q7:
![q7](https://user-images.githubusercontent.com/3770323/39541140-6aaf28a8-4e0a-11e8-8897-71ed615ce262.png)


Before Q11:
![beforeq11](https://user-images.githubusercontent.com/3770323/39541139-6aa284cc-4e0a-11e8-963a-b33f48a58e6a.png)

**Comment by jmbejara** [comment]
With respect to multiindexing, you can do this:

![q21_1](https://user-images.githubusercontent.com/3770323/39541469-735baf02-4e0b-11e8-80fa-3c112e4944bc.png)

![q21_2](https://user-images.githubusercontent.com/3770323/39541476-770b6fac-4e0b-11e8-992a-78370e7568b5.png)

**Comment by jmbejara** [comment]
To change the order of the columns, I am doing it manually like this:

![image](https://user-images.githubusercontent.com/3770323/39541515-99a639ca-4e0b-11e8-825b-c08fa4f2b809.png)

**Comment by afgong** [comment]
Yea, our summary statistics are diverging very mildly at Q7...will go back and double check what's going on. This is what I have right now from Q4 and Q5, respectively:

![screen shot 2018-05-02 at 13 28 13](https://user-images.githubusercontent.com/37759075/39541917-bd69afe4-4e0c-11e8-9d2c-23d3358ad53a.png)

![screen shot 2018-05-02 at 13 28 28](https://user-images.githubusercontent.com/37759075/39541921-bfa94dd2-4e0c-11e8-93f6-6a6606fc8967.png)

**Comment by jmbejara** [comment]
Everything here looks good to me. hmm. I don't know. What are you getting for `df.describe()` at the point at which it diverges?

**Comment by afgong** [comment]
Q7:

![screen shot 2018-05-02 at 13 56 17](https://user-images.githubusercontent.com/37759075/39543421-c95d2a66-4e10-11e8-8434-80ee5b834846.png)

**Comment by jmbejara** [comment]
I think you need to rerun your code from the beginning. My Q7 real_wage max is much larger. It looks like you have already dropped the observations described in Q10 at this point.

**Comment by Jacob-Bishop** [comment]
What is the employment variable supposed to measure? My assumption was LABFORCE, but we dropped that variable earlier on in the code.

**Comment by jmbejara** [comment]
employment was created from the variable `in_labor_force`.  This is because LABFORCE was a variable equal to 0,1, or 2. The variable we created was True or False (1 or 0).

**Comment by Jacob-Bishop** [comment]
Okay. Do you want it to be the average (fraction employed) or the sum (total employed?).

**Comment by afgong** [comment]
@jmbejara hmm, I don't know what's going on. At which question(s) did you drop the missing values? I dropped (`df.dropna(axis=0, how='any')`) at the end of my code at Q4 OR at the beginning of my code at Q5.

**Comment by jmbejara** [comment]
@Jacob-Bishop I was looking for the fraction. Also, be sure to take a weighted average.

**Comment by jmbejara** [comment]
@afgong Sorry for this confusion. I have updated my code so that it only drops missing values at the specific points where I say to drop them in the problem descriptions. This changes my Q7 describe to the following:
![q7_df_describe](https://user-images.githubusercontent.com/3770323/39553771-dc3b9bee-4e34-11e8-82a9-3b785212bc90.png)


At this point, I only drop rows at the end of Q6, calling `df = df.dropna()`

Sorry about this. If your answers look reasonably close, I wouldn't worry to much about this. I've instructed Philip to be generous with the grading in this regard. (Also, it's been interesting to me how little things like this can make replication so challenging.)
